servers/quarkus-server/src/main/java/org/projectnessie/server/providers/DynamoVersionStoreFactory.java

```diff
@@ -34,31 +35,31 @@
 @Dependent
 public class DynamoVersionStoreFactory implements VersionStoreFactory {
   private final DynamoDbClient dynamoDbClient;
+  private final QuarkusDatabaseAdapterConfig config;
 
   /** Creates a factory for dynamodb version stores. */
   @Inject
-  public DynamoVersionStoreFactory(DynamoDbClient dynamoDbClient) {
+  public DynamoVersionStoreFactory(
+      DynamoDbClient dynamoDbClient, QuarkusDatabaseAdapterConfig config) {
     this.dynamoDbClient = dynamoDbClient;
+    this.config = config;
   }
 
   @Override
   public <VALUE, METADATA, VALUE_TYPE extends Enum<VALUE_TYPE>>
       VersionStore<VALUE, METADATA, VALUE_TYPE> newStore(
           StoreWorker<VALUE, METADATA, VALUE_TYPE> worker, ServerConfig serverConfig) {
 
+    DynamoDatabaseClient client = new DynamoDatabaseClient();
```

Comment:
The `DynamoDatabaseClient` (and similar for the other store types) would be clearer managed via Quarkus by a `@Producer` bean and then inject `DynamoDatabaseClient` into this class. WDYT?

Reply:
If we had that, quarkus would create all types of "client" objects all the time (for all backends), would it not? This way we only create the one selected by configuration.

Reply:
Apparently I was wrong :facepalm: unused beans are not instantiated... however, `MongoClient` is always instantiated for some reason :thinking:

Reply:
I'll refactor this tomorrow

Reply:
> however, MongoClient is always instantiated for some reason

That sounds like a Quarkus bug. IIRC, Quarkus 2.2.3 has some fixes around Mongo, maybe that's been fixed as well.

Reply:
Added client singletons in 90b8280422f3dab555be727df30151ec99be8856